composable_kernel/include/tensor_operation/blockwise_gemm_xdlops.hpp

```diff
@@ -363,165 +258,92 @@ struct BlockwiseGemmXdlops_km_kn_m0m1m2n_v1_2x2pipeline
 
-            // read B_sub_0
-            b_thread_copy_.Run(BBlockDesc{},
-                               make_tuple(k, I0, I0, I0),
-                               b_block_buf,
-                               b_thread_desc_,
-                               make_tuple(I0, I0, I0, I0),
-                               b_thread_buf);
+        vector_type<FloatAB, K1> b_thread_vec;
 
-            // C_sub_11 += transpose(A_sub_1) * B_sub_1
-            xdlops_gemm.template Run<decltype(a_thread_desc_),
-                                     decltype(b_thread_desc_),
-                                     decltype(c_thread_desc_),
-                                     1,
-                                     1>(a_thread_buf, b_thread_buf, c_thread_buf);
+        static_for<0, KPerBlock, xdlops_gemm.KPerXdlops>{}([&](auto k) {
+            // read A
+            a_thread_copy_.Run(a_k0_m0_m1_m2_k1_block_desc,
+                               make_tuple(k, I0, I0, I0, I0),
+                               a_block_buf,
+                               a_thread_desc_,
+                               make_tuple(I0, I0, I0, I0, I0),
+                               a_thread_buf);
 
-            // read B_sub_1
-            b_thread_copy_.Run(BBlockDesc{},
-                               make_tuple(k, I1, I0, I0),
+            // read B
+            b_thread_copy_.Run(b_k0_n0_n1_n2_k1_block_desc,
+                               make_tuple(k, I0, I0, I0, I0),
                                b_block_buf,
                                b_thread_desc_,
-                               make_tuple(I0, I1, I0, I0),
+                               make_tuple(I0, I0, I0, I0, I0),
                                b_thread_buf);
 
-            // read A_sub_1
-            a_thread_copy_.Run(ABlockDesc{},
-                               make_tuple(k, I1, I0, I0),
-                               a_block_buf,
-                               a_thread_desc_,
-                               make_tuple(I0, I1, I0, I0),
-                               a_thread_buf);
+            using mfma_input_type =
+                typename vector_type<FloatAB, xdlops_gemm.mfma_type.k_base>::type;
 
-            // C_sub_00 += transpose(A_sub_0) * B_sub_0
-            xdlops_gemm.template Run<decltype(a_thread_desc_),
-                                     decltype(b_thread_desc_),
-                                     decltype(c_thread_desc_),
-                                     0,
-                                     0>(a_thread_buf, b_thread_buf, c_thread_buf);
-
-            // C_sub_01 += transpose(A_sub_0) * B_sub_1
-            xdlops_gemm.template Run<decltype(a_thread_desc_),
-                                     decltype(b_thread_desc_),
-                                     decltype(c_thread_desc_),
-                                     0,
-                                     1>(a_thread_buf, b_thread_buf, c_thread_buf);
+            static_for<0, MRepeat, 1>{}([&](auto m0) {
```

Comment:
Is this still "2x2 pipelined"? If not, please remove the work from the name of this class ```BlockwiseGemmXdlops_km_kn_m0m1m2n_v1_2x2pipeline```

Reply:
Removed BlockwiseGemmXdlops_km_kn_m0m1m2n_v1_2x2pipeline